### DISCUSSION TITLE
How to animate parametric curve smoothly

### DISCUSSION THREAD

**Discussion by tmelorc** [3b1b/manim#1946] [Q&A]
This is my first post here. So, greetings.

First of all, I'm using `Manim Community v0.17.2`.

I'd like to animate the construction of the following parametric curve:
```
x(t) = np.cos(t) * np.sin(1/4*t)
y(t) =  np.sin(t) * np.sin(1/4*t)
```

Well, the following code simply plot the curve on the scene, and it works like a charm. Observe in the screenshot that the curve is smooth enough.
```
class PlotParametricFunction(Scene):
    
    def func(self, t):
        ''' paramatric curve '''
        return np.array((np.cos(t) * np.sin(1/4*t), np.sin(t) * np.sin(1/4*t), 0))

    def construct(self):
        func = ParametricFunction(self.func, t_range = np.array([0, 8*PI])).set_color(ORANGE)

        self.add(func.scale(3))
        self.wait()
```
![PlotParametricFunction](https://user-images.githubusercontent.com/15181573/209569226-7cc1fab4-bf72-4da3-9f2b-2553c166e7b2.png)

Now, in the following, after looking for similar codes over Internet, I was able to animate the construction, but as we can see the curve is far away from being smooth like the previous one. 

```
class AnimateParametricFunction(Scene):
    
    def func(self, t):
        ''' paramatric curve '''
        return np.array((np.cos(t) * np.sin(1/4*t), np.sin(t) * np.sin(1/4*t), 0))

    def construct(self):
        func = ParametricFunction(self.func, t_range = np.array([0, 8*PI])).set_color(ORANGE)

        t = ValueTracker(0)

        dot = Dot(color=WHITE, radius=0.05)
        self.add(dot)
        dot.add_updater(lambda m: m.move_to(
            np.array([
                3 * np.cos(t.get_value()) * np.sin(1/4*t.get_value()), # .scale(3) ?
                3 * np.sin(t.get_value()) * np.sin(1/4*t.get_value()),
                0
            ])
        ))

        path = TracedPath(dot.get_center, stroke_color=ORANGE, stroke_width=3)

        self.add(path)

        self.play(t.animate.set_value(8*PI), run_time=3, rate_func=linear)

        self.remove(dot)
        self.wait()
```
![AnimateParametricFunction](https://user-images.githubusercontent.com/15181573/209569294-1b39e363-51b8-4673-8fc1-d4360eb282f3.png)

So, I'd like to ask: **what is the best way to animate the construction of a smooth parametric curve?**
Thanks in advance.

**Comment by ynhuruc** [comment]
您好！这是QQ邮箱的自动回复，您的邮件已收到，谢谢！
